src/botPage/view/blockly/blocks/tools/total_profit.js

```diff
@@ -9,5 +10,14 @@ Blockly.Blocks.total_profit = {
         this.setTooltip(translate('Returns the total profit'));
         this.setHelpUrl('https://github.com/binary-com/binary-bot/wiki');
     },
+    onchange: function onchange(ev) {
+        if (!this.workspace || this.isInFlyout || this.workspace.isDragging()) {
+            return;
+        }
+
+        if (ev.type === Blockly.Events.MOVE) {
+            beforeAuthenticate(this);
```

Comment:
I believe it's better to move the logic in `beforeAuthenticate` here as we won't use it elsewhere. This will make it easier to find out what's happening on a `Blockly.Events.MOVE` event.